Form/ChoiceList/PhpcrOdmQueryBuilderLoader.php

```diff
@@ -93,8 +107,13 @@ public function getEntitiesByIds($identifier, array $values)
         $qb = clone $this->queryBuilder;
         $alias = $qb->getPrimaryAlias();
         $where = $qb->andWhere()->orX();
+
         foreach ($values as $val) {
-            $where->same($val, $alias);
+            if (UUIDHelper::isUUID($val) && $this->classMetadata) {
```

Comment:
i would change the order of the checks. and add another check on classMetadata to see if the document is referenceable.